_gsocproposals/2024/proposal_MCnetYODA.md

```diff
@@ -0,0 +1,54 @@
+---
+title: MCnet/YODA - Improved unit testing and coverage reports from CI
+layout: gsoc_proposal
+project: MCnet
+year: 2024
+organization:
+  - UCL
+difficulty: medium
+duration: 175
+mentor_avail: June-October
+---
+
+# Description
+
+YODA is a lightweight library for efficient multi-dimensional histogramming
+and binned data analysis. It provides a lightweight common system for
+Monte Carlo event generator validation analyses, particularly as the
+core histogramming library used by the [Rivet](https://yoda.hepforge.org)
+analysis toolkit.
+
+
+## Task ideas
+
+This project will improve the robustness of the YODA library and
+release system through automated unit-test coverage reports
+by the GitLab continuous integration system, and web-based
+visualization of what code areas need better monitoring.
+
+
+## Expected results and milestones
+
+ * Familiarise with the YODA library;
+ * Familiarise with the YODA unit test system, including current CI scripts;
+ * Hook up existing Familiarise with `gcov` for producing coverage report;
+ * Familiarise and its visualization tools
```

Comment:
```suggestion
 * Familiarise with its visualization tools
```
Do you mean that existing visualization tools in YODA or that the student should pick some visualization library to do the coverage reports?

Reply:
The latter! Sorry, I should have read this more carefully last night before committing. I removed the rogue bullet points, so hopefully it makes sense now.

Reply:
Great, now it's very clear, I think.